packages/purpose-process/src/services/purposeService.ts

```diff
@@ -189,6 +191,15 @@ const retrieveRiskAnalysis = (
   return riskAnalysis;
 };
 
+async function retrieveTenantKind(
+  tenantId: TenantId,
+  readModelService: ReadModelService
+): Promise<TenantKind> {
+  const tenant = await retrieveTenant(tenantId, readModelService);
+  assertTenantKindExists(tenant);
```

Comment:
Here you could embed the check
```
if (!tenant.kind) {
    throw tenantKindNotFound(tenant.id);
 }
```
and remove `assertTenantKindExists`, since `retrieveTenantKind` is already a wrapper, so there's no need to go further with the layers of encapsulation. Also, since you added this new new function, you could apply it to the remaining usages in this file.